src/main/java/net/geforcemods/securitycraft/items/TaserItem.java

```diff
@@ -73,8 +79,28 @@ else if(player.inventory.hasItemStack(oneRedstone))
 					return ActionResult.resultPass(stack);
 				}
 
-				WorldUtils.addScheduledTask(world, () -> world.addEntity(new TaserBulletEntity(world, player, powered)));
+				int range = 11;
+				Vector3d startVec = player.getEyePosition(1.0F);
+				Vector3d lookVec = player.getLook(1.0F).scale(range);
+				Vector3d endVec = startVec.add(lookVec);
+				AxisAlignedBB boundingBox = player.getBoundingBox().expand(lookVec).grow(1, 1, 1);
+				EntityRayTraceResult entityRayTraceResult = ProjectileHelper.rayTraceEntities(player, startVec, endVec, boundingBox, s -> s instanceof LivingEntity, range * range);
+
 				SecurityCraft.channel.send(PacketDistributor.ALL.noArg(), new PlaySoundAtPos(player.getPosX(), player.getPosY(), player.getPosZ(), SCSounds.TASERFIRED.path, 1.0F, "players"));
+				if (entityRayTraceResult != null)
+				{
+					LivingEntity entity = (LivingEntity)entityRayTraceResult.getEntity();
+
+					if(!entity.isActiveItemStackBlocking() && entity.attackEntityFrom(CustomDamageSources.TASER, powered ? 2.0F : 1.0F))
+					{
+						int strength = powered ? 4 : 1;
+						int length = powered ? 400 : 200;
+
+						entity.addPotionEffect(new EffectInstance(Effect.get(18), length, strength));
+						entity.addPotionEffect(new EffectInstance(Effect.get(9), length, strength));
+						entity.addPotionEffect(new EffectInstance(Effect.get(2), length, strength));
```

Comment:
```suggestion
						entity.addPotionEffect(new EffectInstance(Effects.WEAKNESS, length, strength));
						entity.addPotionEffect(new EffectInstance(Effects.NAUSEA, length, strength));
						entity.addPotionEffect(new EffectInstance(Effects.SLOWNESS, length, strength));
```
I know you just coped the code over from TaserBulletEntity, but might as well change this in one go.